cfme/tests/integration/test_cfme_auth.py

```diff
@@ -275,3 +277,68 @@ def test_login_local_group(appliance, local_users, local_group, soft_assert):
             soft_assert(local_group.description.lower() in display_groups,
                         'local group "{}" not displayed in UI groups list "{}"'
                         .format(local_group.description, display_groups))
+
+
+@pytest.mark.tier(1)
+@pytest.mark.ignore_stream('5.8')
+@pytest.mark.uncollectif(lambda auth_mode: auth_mode == 'amazon',
+                         'Amazon auth_data needed for group switch testing')
+def test_user_group_switching(appliance, auth_user_data, auth_mode, user_type, auth_provider,
+                              soft_assert, request):
+    """Test switching groups on a single user, between retreived group and built-in group"""
+    UserGroups = namedtuple("UserGroups", "user, retrieved, groups")
+    user_group_tuples = [
+        UserGroups(
+            appliance.collections.users.simple_user(
+                user.username.replace(' ', '-') if user_type == 'upn' else user.username,
+                credentials[user.password]['password'],
+                fullname=user.fullname),
+            retrieve_group(appliance, auth_mode, user.username, group, auth_provider),
+            user.groups
+        )
+        for user in auth_user_data
+        for group in user.groups
+        # pick non-evm group when there are multiple groups for the user
+        if 'evmgroup' not in group.lower() and len(user.groups) > 1]
```

Comment:
**OPTIONAL**
It's not the best using of the list comprehension. I would make regular nested `for` loops.

Reply:
I would agree here, it is quite the list comprehension. if you do not want to change it, maybe add some comments so the next tester does not spend a lot of time on this.

Reply:
You guys make very good points, I originally avoided it because of the level of nesting required. Updating it now, but be warned it nests like 5 times :)